napari/layers/_vectors_layer/model.py

```diff
@@ -66,62 +64,32 @@ def __init__(
         self._need_visual_update = False
 
         # assign vector data and establish default behavior
-        self._raw_data = None
-        self._original_data = vectors
-        self._current_data = vectors
+        self._raw_data = vectors
+        with self.freeze_refresh():
+            self.data = vectors
 
-        self._vectors = self._convert_to_vector_type(vectors)
-        vertices, triangles = self._generate_meshes(self._vectors, self.width)
-        self._mesh_vertices = vertices
-        self._mesh_triangles = triangles
-
-        if name is None:
-            self.name = 'vectors'
-        else:
-            self.name = name
-
-    # ====================== Property getter and setters =====================
     @property
-    def _original_data(self) -> np.ndarray:
-        return self._raw_data
-
-    @_original_data.setter
-    def _original_data(self, data: np.ndarray):
-        """Must preserve data used at construction. Specifically for default
-        averaging/length adjustments.
-        averaging/length adjustments recalculate the underlying data
-
-        Parameters
-        ----------
-        data : np.ndarray
-        """
-        if self._raw_data is None:
-            self._raw_data = data
+    def data(self) -> np.ndarray:
+        return self._data
 
-    @property
-    def vectors(self) -> np.ndarray:
-        return self._vectors
-
-    @vectors.setter
-    def vectors(self, vectors: np.ndarray):
-        """Can accept two data types:
-            1) (N, 4) array with elements (y, x, v, u),
-                where x-y are position (center) and u-v are x-y projections of
-                    the vector
-            2) (N, M, 2) array with elements (v, u)
-                where u-v are x-y projections of the vector
-                vector position is one per-pixel in the NxM array
+    @data.setter
+    def data(self, vectors: np.ndarray):
+        """(N, 2, D) or (N1, N2, ..., ND, D) array: a (N, 2, D) array is
+        interpreted as "coordinate-like" data and a list of N vectors with start
+        point and projections of the vector in D dimensions. A
+        (N1, N2, ..., ND, D) array is interpreted as "image-like" data where
+        there is a length D vector of the projections at each pixel.
 
         Parameters
         ----------
-        vectors : np.ndarray
+        vectors : (N, 2, D) array
         """
-        self._original_data = vectors
-        self._current_data = vectors
 
-        self._vectors = self._convert_to_vector_type(self._current_data)
+        self._data = self._convert_to_vector_type(vectors)
 
-        vertices, triangles = self._generate_meshes(self._vectors, self.width)
+        vertices, triangles = self._generate_meshes(
+            self._data, self.width, self.length
+        )
```

Comment:
Total side note: how hard would it be to make arrows instead of lines? It's kinda weird that the visual is not oriented in any way.

Reply:
it's easy to make arrowheads (that can be toggled on / off), but lets wait for a followup PR